src/sys/mman.rs

```diff
@@ -603,3 +603,64 @@ pub fn shm_unlink<P: ?Sized + NixPath>(name: &P) -> Result<()> {
 
     Errno::result(ret).map(drop)
 }
+
+/// Creates and returns a new, or returns an existing, System V shared memory
+/// segment identifier.
+///
+/// For more information, see [`shmget(2)`].
+///
+/// [`shmget(2)`]: https://man7.org/linux/man-pages/man2/shmget.2.html
+pub fn shmget(key: key_t, size: size_t, shmflg: c_int) -> Result<c_int> {
```

Comment:
`shmflg` should be an enum of the valid values, not a raw C integer. I believe that there is a special macro in nix (`libc_enum!` or something like that) that will streamline the creation of this.